ppa/archive/management/commands/index.py

```diff
@@ -0,0 +1,154 @@
+'''
+**index** is a custom manage command to index PPA digitized work and page
+content into Solr.  It should be run _after_ content has been imported
+to the project database via the **hathi_import** manage command.
+
+Page content will be indexed from a local copy of dataset files under the
+configured **HATHI_DATA** path in pairtree format retrieved by rsync.
+(Note that pairtree data must include pairtree version file to be valid.)
+
+By default, indexes all content, digitized works and pages both.  You
+can optionally specify works or pages only, or index specific items
+by hathi id, but the id must exist in the database and the pairtree content
+for the items still must exist at the configured path.
+
+A progress bar will be displayed by default if there are more than 5
+items to process.  This can be suppressed via script options
+
+Example usage::
+
+    # index everything
+    python manage.py index
+    # index specific items
+    python manage.py index htid1 htid2 htid3
+    # index works only (skip pages)
+    python manage.py index -i works
+    # index pages only (skip works)
+    python manage.py index -i pages
+    # suppress progressbar
+    python manage.py index --no-progress
+
+'''
+
+
+import itertools
+
+from django.core.management.base import BaseCommand, CommandError
+from django.core.paginator import Paginator
+from django.db.models import Sum
+import progressbar
+# from urllib3.exceptions import HTTPError
+from SolrClient.exceptions import ConnectionError
+from requests.exceptions import RequestException
+
+from ppa.archive.models import DigitizedWork
+from ppa.archive.solr import get_solr_connection
+
+
+class Command(BaseCommand):
+    '''Reindex digitized items into Solr that have already been imported'''
+    help = __doc__
+
+    solr = None
+    solr_collection = None
+    bib_api = None
+    hathi_pairtree = {}
+    stats = None
+
+    options = {}
+    #: normal verbosity level
+    v_normal = 1
+    verbosity = v_normal
+
+    #: solr params for index call; currently set to commit within 5 seconds
+    solr_index_opts = {"commitWithin": 5000}
+    # NOTE: not sure what is reasonable here, but without some kind of commit,
+    # Solr seems to quickly run out of memory
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'source_ids', nargs='*',
+            help='Optional list of specific works to index by source (HathiTrust) id.')
+        parser.add_argument(
+            '-i', '--index', choices=['all', 'works', 'pages'],  default='all',
+            help='Index only works or pages (by default indexes all)')
+        parser.add_argument(
+            '--no-progress', action='store_true',
```

Comment:
👍  for defaulting to progressbar :)

Reply:
@thatbudakguy inspired by your feedback on the import script.

There's no summary output in this one - should there be?

Reply:
I don't think a summary output is necessary, since we can ping the solr server for the outcome and number of docs it currently has, etc. The progressbar is more meaningful in day to day use imo.

If it's a quick add though, more info is better than less.